### PR TITLE
Add the template lookup escaping to the 2.4 porting guide

### PR DESCRIPTION
**Pull request by abadger**
##### SUMMARY

This change can break playbooks so make sure we include it in the porting guide.

##### ISSUE TYPE
 - Docs Pull Request

##### ANSIBLE VERSION

```
devel and 2.4
```

### CODE REVIEW

File: docs/docsite/rst/porting_guide_2.4.rst [hunk 1]
```diff
@@ -131,6 +131,28 @@ Developers:
 * Any callbacks inheriting from other callbacks might need to also be updated to contain the same documented options
   as the parent or the options won't be available.  This is noted in the developer guide.
 
+Template lookup plugin
+----------------------
+
+Prior to Ansible 2.4, the strings passed to the template lookup plugin would have backslashes
```
Comment (gundalow) [inline docs/docsite/rst/porting_guide_2.4.rst]: >template lookup plugin would have backslashes
Would have -> must have?

File: docs/docsite/rst/porting_guide_2.4.rst [hunk 2]
```diff
@@ -131,6 +131,28 @@ Developers:
 * Any callbacks inheriting from other callbacks might need to also be updated to contain the same documented options
   as the parent or the options won't be available.  This is noted in the developer guide.
 
+Template lookup plugin
```
Comment (gundalow) [inline docs/docsite/rst/porting_guide_2.4.rst]: Can we give more or a clue in the heading, perhaps?
`Template lookup plugin: Escaping strings`

File: docs/docsite/rst/porting_guide_2.4.rst [hunk 3]
```diff
@@ -131,6 +131,28 @@ Developers:
...
+escaped.  This has changed so that users have to escape backslashes themselves in 2.4.  This change
+brings the template lookup plugin inline with the template module so that the same backslash
+escaping rules apply to both.
+
+If you have a template lookup like this::
+
+    - debug:
+        msg: '{{ lookup("template", "tpl.j2") }}'
+
+And the template in :file:`tpl.j2` is::
```
Comment (gundalow) [inline docs/docsite/rst/porting_guide_2.4.rst]: tpl-> template?

File: docs/docsite/rst/porting_guide_2.4.rst [hunk 4]
```diff
@@ -131,6 +131,28 @@ Developers:
...
+
+    {{ "name surname" | regex_replace("^[^\s]+\s+(.*)", "\1") }}
+
+Change the template like this::
```
Comment (gundalow) [inline docs/docsite/rst/porting_guide_2.4.rst]: in the `2.3` guide we used the form
OLD In Ansible 2.2 (and earlier)
NEW In Ansible 2.3:

https://github.com/abadger/ansible/blob/devel/docs/docsite/rst/porting_guide_2.3.rst#openbsd-version-facts

**Review by gundalow — CHANGES_REQUESTED** [review]

**Review by gundalow — APPROVED** [review]

**Review by dharmabumstead — APPROVED** [review]

**Review by dharmabumstead — APPROVED** [review]
Approved. Thanks @abadger

**Pull request merged**